### PR TITLE
Move some codegen-y methods from `rustc_hir_analysis::collect` -> `rustc_codegen_ssa`

### PR DESCRIPTION
**Pull request by compiler-errors**
Unclear if they should live here, but they seem codegen-y enough, and `rustc_hir_analysis::collect` is extremely long, so it should probably lose some methods.

### CODE REVIEW

File: compiler/rustc_codegen_ssa/src/target_features.rs [hunk 1]
```diff
@@ -322,15 +333,148 @@ pub fn tied_target_features(sess: &Session) -> &'static [&'static [&'static str]
...
-    providers.supported_target_features = |tcx, cnum| {
-        assert_eq!(cnum, LOCAL_CRATE);
-        if tcx.sess.opts.actually_rustdoc {
-            // rustdoc needs to be able to document functions that use all the features, so
-            // whitelist them all
-            all_known_features().map(|(a, b)| (a.to_string(), b)).collect()
-        } else {
-            supported_target_features(tcx.sess).iter().map(|&(a, b)| (a.to_string(), b)).collect()
-        }
+pub fn from_target_feature(
```
Comment (estebank) [inline compiler/rustc_codegen_ssa/src/target_features.rs:336]: Why is this function defined twice?

Comment (compiler-errors) [reply compiler/rustc_codegen_ssa/src/target_features.rs:336]: Done, should (probably) not have any more duplicates.

**Comment by estebank** [comment]
r=me

**Pull request merged**